### ISSUE TITLE
Async function return void cause `invalid void`

### ISSUE THREAD

**Issue by ruowan** [palantir/tslint#4828]
### Bug Report

- __TSLint version__: 5.19.0
- __TypeScript version__: 3.5.3
- __Running TSLint via__: CLI / VSCode 
#### TypeScript code being linted

```ts
const sayHello = async (): Promise<void> => {
  // do something
}
```

with `tslint.json` configuration:

```json
default
```

#### Actual behavior
After install tslint@5.19.0, tslint reports it. In previous version tslint doesn't report this error.

tslint report 
```
void is not a valid type other than return types (invalid-void)tslint(1)
```
#### Expected behavior
I think `async function return  Promise<void>` is general usage.  tslint shouldn't report the error.
[invalid-void doc](https://palantir.github.io/tslint/rules/invalid-void/) seems don't contain this usage.

**Comment by JoshuaKGoldberg** [comment]
> `Promise<void>`

Sure, that seems like a common use case.

How about we add an option to the rule to allow generics to be `void`. It should allow either `true` for all generics or a `string[]` whitelist of allowed ones?

```json
{
    "allow-generics": true
}
```

```json
{
    "allow-generics": ["Promise"]
}
```

**Comment by ruowan** [comment]
That's a good idea to solve this problem.  So the `invalid-void` would be like this? Does`allow-generics` means whitelist for generics? 

```
{
  "invalid-void": {
      true;
     "allow-generics": true
  }
}
```

I have another question. why this problem only occurs in version 5.19.0.  Is there any change in this rule? Thanks.

**Comment by JoshuaKGoldberg** [comment]
> Does `allow-generics` means whitelist for generics?

Yup!

> why this problem only occurs in version 5.19.0. Is there any change in this rule? Thanks.

The `invalid-void` rule was added in 5.19.0: https://github.com/palantir/tslint/releases/tag/5.19.0. If you're using `tslint:all` to take in rules, this would have been enabled for you.

**Comment by michaelw85** [comment]
Following, running into the same issue with return type of `Promise<void>`
